backend/backend/metric.py

```diff
@@ -0,0 +1,155 @@
+from collections import Counter
+import numpy as np
+
+from classification.config.constants import SleepStage, EPOCH_DURATION
+
+INVALID_TIMESTAMP = -1
+
+
+class Metrics():
+    def __init__(self, sleep_stages, bedtime):
+        self.sleep_stages = sleep_stages
+        self.bedtime = bedtime
+        self.has_slept = len(np.unique(self.sleep_stages)) != 1 or np.unique(self.sleep_stages)[0] != SleepStage.W.name
+
+        self.is_sleeping_stages = self.sleep_stages != SleepStage.W.name
+        self.sleep_indexes = np.where(self.is_sleeping_stages)[0]
+
+        self._initialize_sleep_offset()
+        self._initialize_sleep_latency()
+        self._initialize_rem_latency()
+        self._initialize_transition_based_metrics()
+
+    @property
+    def report(self):
+        report = {
+            'sleepOffset': self._sleep_offset,
+            'sleepLatency': self._sleep_latency,
+            'remLatency': self._rem_latency,
+            'awakenings': self._awakenings,
+            'stageShifts': self._stage_shifts,
+            'sleepTime': self._sleep_time,
+            'WASO': self._wake_after_sleep_onset,
+            'sleepEfficiency': self._sleep_efficiency,
+            'efficientSleepTime': self._efficient_sleep_time,
+            'wakeAfterSleepOffset': self._wake_after_sleep_offset,
+            'sleepOnset': self._sleep_onset,
+            'remOnset': self._rem_onset,
+            **self._time_passed_in_stage,
+        }
+
+        for metric in report:
+            #  json does not recognize NumPy data types
+            if isinstance(report[metric], np.int64):
+                report[metric] = int(report[metric])
```

Comment:
is this because of shape[0]?

Reply:
Now that I've changed shape[0] to len, I still got these metrics in np.int64:
```
Int64 var:  sleepOffset
Int64 var:  sleepLatency
Int64 var:  remLatency
Int64 var:  sleepTime
Int64 var:  WASO
Int64 var:  wakeAfterSleepOffset
Int64 var:  sleepOnset
Int64 var:  remOnset
```
I think it's better to ensure afterwards they're all int type instead of np.int64, as the reason is more explicit.